### PR TITLE
[core] Clean up technical debt in defaulting code (1)

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area dev-productivity
/kind technical-debt

**What this PR does / why we need it**:

This PR cleans up technical debt in our API defaulting code according to https://github.com/gardener/gardener/issues/7312.
This PR tackles the following resources in the `core` API group:
- `Project`
- `CloudProfile`
- `ControllerRegistration`

**Which issue(s) this PR fixes**:
Part of https://github.com/gardener/gardener/issues/7312

**Special notes for your reviewer**:

Use the dedicated commits to review this PR step-by-step.
Commits named `Prefactor: extract defaulting into dedicated file` don't change code or tests but only move them for easier reviews of the actual refactoring

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: pkg/apis/core/v1beta1/defaults_cloudprofile_test.go [hunk 1]
```diff
@@ -17,39 +17,53 @@ package v1beta1_test
...
-		})
+	BeforeEach(func() {
+		obj = &CloudProfile{
+			Spec: CloudProfileSpec{
+				MachineImages: []MachineImage{{
+					Name: "test",
+					Versions: []MachineImageVersion{{
+						ExpirableVersion: ExpirableVersion{
+							Version: "v1.0.0",
+						},
```
Comment (timuthy) [inline pkg/apis/core/v1beta1/defaults_cloudprofile_test.go]: Any reason why this is set now? Technically speaking, it changes the test.

File: pkg/apis/core/v1beta1/defaults_cloudprofile_test.go [hunk 2]
```diff
@@ -17,39 +17,53 @@ package v1beta1_test
...
+					}},
+				}},
+				MachineTypes: []MachineType{{
+					Name: "large",
```
Comment (timuthy) [inline pkg/apis/core/v1beta1/defaults_cloudprofile_test.go]: Same as comment above.

File: pkg/apis/core/v1beta1/defaults_cloudprofile.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	if obj.Usable == nil {
+		obj.Usable = pointer.Bool(true)
+	}
+}
+
+// SetDefaults_VolumeType sets default values for VolumeType objects.
+func SetDefaults_VolumeType(obj *VolumeType) {
+	if obj.Usable == nil {
+		trueVar := true
+		obj.Usable = &trueVar
```
Comment (timuthy) [inline pkg/apis/core/v1beta1/defaults_cloudprofile.go]:
```suggestion
                obj.Usable = pointer.Bool(true)
```

File: pkg/apis/core/v1alpha1/defaults_controllerregistration.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+			obj.GloballyEnabled = pointer.Bool(false)
+		}
+
+		if obj.ReconcileTimeout == nil {
+			obj.ReconcileTimeout = &metav1.Duration{Duration: time.Minute * 3}
+		}
+
+		if obj.Lifecycle == nil {
+			obj.Lifecycle = &ControllerResourceLifecycle{}
+		}
```
Comment (timuthy) [inline pkg/apis/core/v1alpha1/defaults_controllerregistration.go:41]: Why is this defaulted?

**Review by timuthy — COMMENTED** [review]
Very nice PR, I very appreciate this refactoring 🙂

File: pkg/apis/core/v1beta1/defaults_cloudprofile_test.go (hunk 1 above)
Comment (timebertt) [reply pkg/apis/core/v1beta1/defaults_cloudprofile_test.go]: Right. I just thought I could make it a bit more realistic by specifying the required fields.
It doesn't really matter, so I removed the `name` and `version` fields again.

File: pkg/apis/core/v1beta1/defaults_cloudprofile_test.go (hunk 2 above)
Comment (timebertt) [reply pkg/apis/core/v1beta1/defaults_cloudprofile_test.go]: Same as answer above :)

File: pkg/apis/core/v1alpha1/defaults_controllerregistration.go (hunk 4 above)
Comment (timebertt) [reply pkg/apis/core/v1alpha1/defaults_controllerregistration.go:41]: What do you mean? This was also defaulted before the refactoring: 
https://github.com/gardener/gardener/blob/2868508f994821c68fc37289aa97b49d43b856dc/pkg/apis/core/v1alpha1/defaults.go#L479-L481

**Comment by timebertt** [comment]
@timuthy thanks for the review! PTAL :)

**Review by timuthy — COMMENTED** [review]
/lgtm
/approve

**Comment by timuthy** [comment]
@shafeeqes I leave the decision up to you to include it in the `v1.63` release. The change itself is not important for the release, though.

**Pull request merged**